src/execution/compiler/operator/static_aggregation_translator.cpp

```diff
@@ -161,6 +160,19 @@ void StaticAggregationTranslator::UpdateGlobalAggregate(WorkContext *ctx, Functi
   }
 }
 
+ast::Expr *StaticAggregationTranslator::GetValuesStructTypeForTerm(
+    const common::ManagedPointer<parser::AggregateExpression> &term) {
+  auto codegen = GetCodeGen();
+  if (term->GetExpressionType() == parser::ExpressionType::AGGREGATE_AVG) {
+    // No guarantee that we have derived the return type of the child
+    // expression at this point in query processing, so we need to do it here.
+    TERRIER_ASSERT(term->GetChildrenSize() >= 1, "No column name given.");
+    const_cast<parser::AbstractExpression *>(term->GetChild(0).Get())->DeriveReturnValueType();
```

Comment:
I would actually be quite concerned if we need to derive return value type all the way here, this should be the binder and/or optimizer's job. I would prefer an assert saying that the type isn't INVALID.

Reply:
Good point. I neglected to consider that we necessarily `Visit()` all of the children of the aggregate expression in the plan before we get to this step in processing. I wonder if that means we can remove a similar invocation in the implementation of `DeriveReturnValueType()` for `AggregateExpression` because we can make a similar assumption? I'll make the update.